# Non Graded Challenge 5

# Linear Algebra (Eigen Decomposition)

Nama : Raditya Saputra

Batch : 007 FTDS

# Library yang digunakan numpy

In [45]:
import numpy as np
from numpy.linalg import eig

# Soal nomor 1

- Apakah $\begin{bmatrix} 1 \ 3 \end{bmatrix}$ adalah eigenvector dari $\begin{bmatrix} 1 \ -1 \ 6 \ 4 \end{bmatrix}$ Jika iya, berapakah eigenvalue-nya?

In [19]:
A = np.array ([[1, -1], [6, 4]])

In [20]:
A

array([[ 1, -1],
       [ 6,  4]])

In [21]:
A.shape

(2, 2)

# Eigen Values dan Eigen Vector

In [22]:
values, vectors = eig (A)

In [23]:
print (values)
print (vectors)

[2.5+1.93649167j 2.5-1.93649167j]
[[-0.23145502+0.29880715j -0.23145502-0.29880715j]
 [ 0.9258201 +0.j          0.9258201 -0.j        ]]


# Rekontruksi matriks

In [24]:
from numpy.linalg import eig,inv
from numpy import diag

In [25]:
rekontruksi_matriks = vectors.dot(diag(values)).dot(inv(vectors))

In [26]:
rekontruksi_matriks

array([[ 1.+0.j, -1.+0.j],
       [ 6.+0.j,  4.+0.j]])

In [27]:
eigval, eigvec = eig (A)

In [28]:
eigval

array([2.5+1.93649167j, 2.5-1.93649167j])

In [35]:
eigvec

array([[-0.23145502+0.29880715j, -0.23145502-0.29880715j],
       [ 0.9258201 +0.j        ,  0.9258201 -0.j        ]])

In [43]:
eig_vec = np.array ([[1],[3]])

Pembuktiaan eigen vector pada matriks

In [34]:
eig_vec 

array([[1],
       [3]])

In [44]:
np.sqrt (320)

17.88854381999832

In [42]:
eig_vec / np.sqrt (320)

array([[0.0559017],
       [0.1677051]])

# Overall Analysis

apakah eigen vector ([[1],[3]]) dari matriks ([[1, -1], [6, 4]]) : tidak
eigen values : ([2.5+1.93649167j, 2.5-1.93649167j]) bilangan imajiner, apabila eigen values berisikan bilangan imajiner, berarti matriks tersebut tidak ada eigen vectors.